src/patient-registration/input/custom-input/identifier/identifier-input.component.tsx

```diff
@@ -98,7 +101,7 @@ export const IdentifierInput: React.FC<IdentifierInputProps> = ({
               : 'Enter identifier'
           }
           name={name}
-          disabled={!option.manualEntryEnabled}
+          disabled={!option.manualEntryEnabled || (field.value && inEditMode) ? true : false}
```

Comment:
Nice catch, I didn't realise this actually. I think the best way to handle this is by not rendering the identifier inputs components for identifier types with identifier values on the UI just like it is in the Ref App since we don't want to edit patient identifiers.